# Títulos de Netflix
## Búsqueda de datos faltantes y tipos de datos

En este análisis veremos cómo detectar valores faltantes (`NaN`) en un DataFrame y cómo interpretar los tipos de datos de cada columna.

In [1]:
import pandas as pd

Leemos el archivo que contiene información sobre títulos disponibles en Netflix:

In [2]:
netflix = pd.read_csv("data/titulos_netflix.csv")
netflix

,show_id,tipo,título,director,año_lanzamiento,clasificación,duración
0,s1,Movie,Dick Johnson Is Dead,Kirsten Johnson,2020,PG-13,90 min
1,s2,TV Show,Blood & Water,NaN,2021,TV-MA,2 Seasons
2,s3,TV Show,Ganglands,Julien Leclercq,2021,TV-MA,1 Season
3,s4,TV Show,Jailbirds New Orleans,NaN,2021,TV-MA,1 Season
4,s5,TV Show,Kota Factory,NaN,2021,TV-MA,2 Seasons
...,...,...,...,...,...,...,...
8802,s8803,Movie,Zodiac,David Fincher,2007,R,158 min
8803,s8804,TV Show,Zombie Dumb,NaN,2018,TV-Y7,2 Seasons
8804,s8805,Movie,Zombieland,Ruben Fleischer,2009,R,88 min
8805,s8806,Movie,Zoom,Peter Hewitt,2006,PG,88 min


Quizás a simple vista no podamos notar si hay datos faltantes. En Pandas, los datos ausentes se representan como `NaN`, que significa *Not a Number*.

Para revisar si hay columnas con valores ausentes y conocer el tipo de cada una, usamos el método `info()`:

In [3]:
netflix.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8807 entries, 0 to 8806
Data columns (total 7 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   show_id          8807 non-null   object
 1   tipo             8807 non-null   object
 2   título           8807 non-null   object
 3   director         6173 non-null   object
 4   año_lanzamiento  8807 non-null   int64 
 5   clasificación    8803 non-null   object
 6   duración         8804 non-null   object
dtypes: int64(1), object(6)
memory usage: 481.8+ KB


### Interpretación del resultado
- Se observa que el DataFrame tiene 8807 registros.
- Algunas columnas como `director`, `clasificación` y `duración` tienen **menos de 8807 valores no nulos**, lo que indica que tienen datos faltantes (`NaN`).

## ¿Qué tipo de dato tiene cada columna?
Además de los valores no nulos, `info()` también muestra el **tipo de dato** de cada columna.

- Por ejemplo, la columna `año_lanzamiento` es de tipo `int64`, lo que indica que contiene números enteros.
- Las columnas con tipo `object` suelen contener cadenas de texto (strings).

In [4]:
netflix.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8807 entries, 0 to 8806
Data columns (total 7 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   show_id          8807 non-null   object
 1   tipo             8807 non-null   object
 2   título           8807 non-null   object
 3   director         6173 non-null   object
 4   año_lanzamiento  8807 non-null   int64 
 5   clasificación    8803 non-null   object
 6   duración         8804 non-null   object
dtypes: int64(1), object(6)
memory usage: 481.8+ KB


> ✅ **Nota**: El tipo `object` en Pandas normalmente representa datos de texto.